install_deps.sh

```diff
@@ -132,40 +289,88 @@ then
+        "${deps_log_dir}" \
+        "${script_path_openxr}" \
+        "${parent_dir_openxr}" \
+        "${dep_prompt_openxr}" \
+        "${dep_ver_openxr}"
 
-    if [ ! -d "${opt_dir}/gtsam" ] && y_or_n "Next: Install gtsam from source"; then
-        . ./scripts/install_gtsam.sh
-    fi
+    prompt_install \
+        "${dep_name_gtsam}" \
+        "${deps_log_dir}" \
+        "${script_path_gtsam}" \
+        "${parent_dir_gtsam}" \
+        "${dep_prompt_gtsam}" \
+        "${dep_ver_gtsam}"
 
-    if [ ! -d "${opt_dir}/opengv" ] && y_or_n "Next: Install opengv from source"; then
-        . ./scripts/install_opengv.sh
-    fi
+    prompt_install \
+        "${dep_name_opengv}" \
+        "${deps_log_dir}" \
+        "${script_path_opengv}" \
+        "${parent_dir_opengv}" \
+        "${dep_prompt_opengv}" \
+        "${dep_ver_opengv}"
 
-    if [ ! -d "${opt_dir}/DBoW2" ] && y_or_n "Next: Install DBoW2 from source"; then
-        . ./scripts/install_dbow2.sh
-    fi
+    prompt_install \
+        "${dep_name_dbow2}" \
+        "${deps_log_dir}" \
+        "${script_path_dbow2}" \
+        "${parent_dir_dbow2}" \
+        "${dep_prompt_dbow2}" \
+        "${dep_ver_dbow2}"
 
-    if [ ! -d "${opt_dir}/Kimera-RPGO" ] && y_or_n "Next: Install Kimera-RPGO from source"; then
-        . ./scripts/install_kimera_rpgo.sh
-    fi
+    prompt_install \
+        "${dep_name_kimera_rpgo}" \
+        "${deps_log_dir}" \
+        "${script_path_kimera_rpgo}" \
+        "${parent_dir_kimera_rpgo}" \
+        "${dep_prompt_kimera_rpgo}" \
+        "${dep_ver_kimera_rpgo}"
 
-    if ! which conda 2> /dev/null; then
-        if [ ! -d "$HOME/miniconda3" ]; then
-            if y_or_n "Next: Install Conda"; then
-                . ./scripts/install_conda.sh
-            fi
-        fi
-    fi
+    prompt_install \
+        "${dep_name_conda}" \
+        "${deps_log_dir}" \
+        "${script_path_conda}" \
+        "${parent_dir_conda}" \
+        "${dep_prompt_conda}" \
+        "${dep_ver_conda}"
 
-    # Load new library paths
+    ## Load new library paths
     sudo ldconfig
 
-    # I won't ask the user first, because this is not a global installation.
-    # All of this stuff goes into a project-specific venv.
-    $HOME/miniconda3/bin/conda env create --force -f runner/environment.yml
+    ### Virtual environment creation ###
+
+    echo "Attempting to create a virtual environment configuration via conda ..."
+
+    ## Check for a previous conda installation
+    detect_dependency "${dep_name_conda}" "${deps_log_dir}"
+    if [ "$?" -eq 0 ]; then
+        dep_log_path_conda="${dep_log_path}"
+        src_dir_conda="${src_dir}"
+        echo "Found conda installation log '${dep_log_path_conda}' : dir <- '${src_dir_conda}'"
+    else
+        dep_log_path_conda="${dep_log_path}"
+        src_dir_conda="${parent_dir_conda}/${dep_name_conda}"
+        print_warning "Installation log for conda not found at '${dep_log_path_conda}'."
```

Comment:
Does conda not found imply we should quit?

Reply:
Yeah I think it would be safe to abort here. The script is written to try for conda anyway below as a stop-gap (`detect_dependency` is only looking for a log entry in `.deps/`, but conda can still have been installed via a previous version of the script, or via the system).